iree/compiler/Conversion/LinalgToLLVM/ConvertToLLVM.cpp

```diff
@@ -124,6 +124,182 @@ class RemoveMakeRankedShape : public ConvertToLLVMPattern {
+    auto funcOp = dyn_cast_or_null<FuncOp>(op);
+    FunctionType fnType = funcOp.getType();
+    if (fnType.getNumInputs() != 0)
+      return rewriter.notifyMatchFailure(
+          funcOp, "entry function should not have inputs");
+
+    // Get interface buffers from all the blocks.
+    SmallVector<IREE::PlaceholderOp, 8> bufferOps;
+    SmallVector<IREE::HAL::InterfaceLoadConstantOp, 8> loadOps;
+    for (Block &block : funcOp.getBlocks()) {
+      for (Operation &op : block) {
+        if (auto phOp = dyn_cast<IREE::PlaceholderOp>(op))
+          bufferOps.push_back(phOp);
+        if (auto phOp = dyn_cast<IREE::HAL::InterfaceLoadConstantOp>(op)) {
+          loadOps.push_back(phOp);
+        }
+      }
+    }
+
+    if (bufferOps.empty()) return failure();
+
+    // A map from buffer ops to their corresponding interface binding ops.
+    llvm::DenseMap<Operation *, IREE::HAL::InterfaceBindingOp> bufferBindingMap;
+    for (auto bufferOp : bufferOps) {
+      auto symbol = SymbolTable::lookupNearestSymbolFrom(
+          bufferOp, bufferOp.getAttrOfType<SymbolRefAttr>("binding"));
+      bufferBindingMap[bufferOp] = cast<IREE::HAL::InterfaceBindingOp>(symbol);
+    }
+
+    // Sort buffers according to their descriptor (set, binding) pair.
+    llvm::sort(bufferOps, [&bufferBindingMap](IREE::PlaceholderOp aBuffer,
+                                              IREE::PlaceholderOp bBuffer) {
+      return bufferBindingMap[aBuffer] < bufferBindingMap[bBuffer];
+    });
+
+    // A map from buffer ops to their corresponding function argument indices.
+    llvm::DenseMap<Operation *, unsigned> bufferArgMap;
+    // A map from binding ops to their corresponding function argument indices.
+    llvm::DenseMap<Operation *, unsigned> bindingArgMap;
+    llvm::SmallVector<MemRefType, 4> inputMemRefTypes;
+    llvm::SmallVector<LLVM::LLVMType, 4> inputStructPtrs;
+    unsigned argIndex = 0;
+    for (auto bufferOp : bufferOps) {
+      auto binding = bufferBindingMap[bufferOp];
+      auto it = bindingArgMap.find(binding);
+      if (it != bindingArgMap.end()) {
+        bufferArgMap[bufferOp] = it->second;
+      } else {
+        bindingArgMap[binding] = argIndex;
+        bufferArgMap[bufferOp] = argIndex;
+        ++argIndex;
+      }
+
+      auto memrefType = bufferOp.getType().dyn_cast_or_null<MemRefType>();
+      inputMemRefTypes.push_back(memrefType);
+      auto elementType =
+          unwrap(typeConverter.convertType(memrefType.getElementType()));
+      inputStructPtrs.push_back(
+          elementType.getPointerTo(memrefType.getMemorySpace()));
+    }
+
+    TypeConverter::SignatureConversion signatureConverter(/*numOrigInputs=*/0);
+
+    // func foo(%buffer_args: !llvm<i8**>, %dim_valus: !llvm<i32*>)
+    auto interfaceInputArg =
+        LLVM::LLVMType::getInt8PtrTy(typeConverter.getDialect()).getPointerTo();
+    auto dimValuesArgType =
+        LLVM::LLVMType::getInt64Ty(typeConverter.getDialect()).getPointerTo();
```

Comment:
int64 doesn't match comment above (i32)?

Reply:
Fixed the comment, the runtime push to vector<int64_t> there is a todo to consider move the casting to LLVMIR.